features/facebookImport/src/model/caching.js

```diff
@@ -0,0 +1,80 @@
+import EntityDeserializer from "./entities/utils/entity-serialisation";
+
+const namespace = "http://polypoly.coop/schema/fbImport/";
+
+export default class ExportDataCache {
+    constructor(pod, fileId) {
+        this._pod = pod;
+        this._fileId = fileId;
+    }
+
+    _createAccessQuery(dataFactory) {
+        return {
+            subject: dataFactory.namedNode(`${this._fileId}/cache`),
+            predicate: dataFactory.namedNode(`${namespace}importerCache`),
+        };
+    }
+
+    async _selectDataQuad(dataFactory, polyIn) {
+        return await polyIn.select(this._createAccessQuery(dataFactory));
+    }
+
+    _createDataQuad(dataFactory, rawData) {
+        return dataFactory.quad(
+            dataFactory.namedNode(`${this._fileId}/cache`),
+            dataFactory.namedNode(`${namespace}importerCache`),
+            dataFactory.literal(rawData)
+        );
+    }
+
+    async _readDataQuad(dataFactory, polyIn) {
+        const dataQuads = await this._selectDataQuad(dataFactory, polyIn);
+        if (dataQuads.length === 0) return undefined;
+        return dataQuads[0];
+    }
+
+    async readData() {
+        const { dataFactory, polyIn } = this._pod;
+        const dataQuad = await this._readDataQuad(dataFactory, polyIn);
+        if (!dataQuad) return undefined;
+        const rawData = dataQuad.object?.value;
+        return rawData;
+    }
+
+    async clearData() {
+        const { dataFactory, polyIn } = this._pod;
+        const dataQuad = await this._readDataQuad(dataFactory, polyIn);
+        if (!dataQuad) return undefined;
+        polyIn.delete(dataQuad);
```

Comment:
Does this return something? In that case, does this only return if it can't delete?